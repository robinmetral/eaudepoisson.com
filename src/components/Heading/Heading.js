import React from "react"
import PropTypes from "prop-types"
import styled from "@emotion/styled"
import { css } from "@emotion/core"

const StyledHeading = styled.span`
  ${({ as }) => css`
    font-family: "Merriweather", serif;
    text-align: ${as === "h1" ? "center" : "left"};
  `}
`

// the header component supports levels 1 to 3
const Heading = ({ h, children }) => (
  <StyledHeading as={`h${h}`} data-testid="header">
    {children}
  </StyledHeading>
)

Heading.propTypes = {
  h: PropTypes.oneOf(["1", "2", "3"]).isRequired,
}

export default Heading
