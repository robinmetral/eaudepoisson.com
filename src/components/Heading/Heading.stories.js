import React from "react";
import Heading from "./Heading";

export default {
  title: "Heading",
};

export const h1 = () => <Heading h="1">Heading 1</Heading>;

export const h2 = () => <Heading h="2">Heading 2</Heading>;

export const h3 = () => <Heading h="3">Heading 3</Heading>;
