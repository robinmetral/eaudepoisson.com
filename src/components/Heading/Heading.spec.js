import React from "react"
import { render } from "test-utils"
import { matchers } from "jest-emotion"

import Heading from "./Heading"

// import emotion matchers
expect.extend(matchers)

describe("Heading", () => {
  it("should render the title text", () => {
    const { getByTestId } = render(<Heading h="1">Title</Heading>)

    expect(getByTestId("header")).toHaveTextContent("Title")
  })

  it("should have typeface merriweather", () => {
    const { getByTestId } = render(<Heading h="1">Title</Heading>)

    expect(getByTestId("header")).toHaveStyleRule(
      "font-family",
      '"Merriweather",serif'
    )
  })

  describe("h1", () => {
    it("should match the snapshots", () => {
      const header = render(<Heading h="1">Title</Heading>)

      expect(header).toMatchSnapshot()
    })

    it("should render an h1 element", () => {
      const h1 = render(<Heading h="1">Title</Heading>)

      expect(h1.getByTestId("header")).toContainHTML("<h1")
    })
  })

  describe("h2", () => {
    it("should match the snapshots", () => {
      const header = render(<Heading h="2">Title</Heading>)

      expect(header).toMatchSnapshot()
    })

    it("should render an h2 element", () => {
      const h2 = render(<Heading h="2">Title</Heading>)

      expect(h2.getByTestId("header")).toContainHTML("<h2")
    })
  })

  describe("h3", () => {
    it("should match the snapshots", () => {
      const header = render(<Heading h="2">Title</Heading>)

      expect(header).toMatchSnapshot()
    })

    it("should render an h3 element", () => {
      const h3 = render(<Heading h="3">Title</Heading>)

      expect(h3.getByTestId("header")).toContainHTML("<h3")
    })
  })
})
