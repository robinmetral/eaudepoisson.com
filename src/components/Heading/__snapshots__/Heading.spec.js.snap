// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading h1 should match the snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  font-family: "Merriweather",serif;
  text-align: center;
}

<body>
    <div>
      <h1
        class="emotion-0 emotion-1"
        data-testid="header"
      >
        Title
      </h1>
    </div>
  </body>,
  "container": .emotion-0 {
  font-family: "Merriweather",serif;
  text-align: center;
}

<div>
    <h1
      class="emotion-0 emotion-1"
      data-testid="header"
    >
      Title
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Heading h2 should match the snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  font-family: "Merriweather",serif;
  text-align: left;
}

<body>
    <div>
      <h2
        class="emotion-0 emotion-1"
        data-testid="header"
      >
        Title
      </h2>
    </div>
  </body>,
  "container": .emotion-0 {
  font-family: "Merriweather",serif;
  text-align: left;
}

<div>
    <h2
      class="emotion-0 emotion-1"
      data-testid="header"
    >
      Title
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Heading h3 should match the snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  font-family: "Merriweather",serif;
  text-align: left;
}

<body>
    <div>
      <h2
        class="emotion-0 emotion-1"
        data-testid="header"
      >
        Title
      </h2>
    </div>
  </body>,
  "container": .emotion-0 {
  font-family: "Merriweather",serif;
  text-align: left;
}

<div>
    <h2
      class="emotion-0 emotion-1"
      data-testid="header"
    >
      Title
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
